packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -795,11 +799,22 @@ export class GarbageCollector implements IGarbageCollector {
          * 3. A summary that was generated before GC even existed.
          * 4. A summary that was generated with tombstone feature disabled.
          */
-        if (!this.tombstoneMode || baseSnapshotData?.tombstones === undefined) {
+        if (baseSnapshotData?.tombstones === undefined) {
             return;
         }
-        this.tombstones = Array.from(baseSnapshotData.tombstones);
-        this.runtime.updateTombstonedRoutes(this.tombstones);
+
+        // If running sweep, the tombstone state represents the list of nodes that were deleted during sweep.
+        if (this.shouldRunSweep) {
```

Comment:
Even if shouldRunSweep is false, shouldn't we want to still copy the baseSnapshotData.tombstones?

Reply:
From you comment on the sweep design doc, I think we should write this if GC is enabled irrespective of whether sweep is enabled or not. Writing it when GC is disabled is possible but not sure if it's worth.